src/MulticastService.cs

```diff
@@ -200,61 +198,102 @@ public void Start()
 
         void FindNetworkInterfaces()
         {
-            var nics = GetNetworkInterfaces()
-                .Where(nic => !knownNics.Any(k => k.Id == nic.Id))
-                .ToArray();
-            foreach (var nic in nics)
+            var nics = GetNetworkInterfaces().ToList();
```

Comment:
I prefer `ToArray`.  Its more performant.

Reply:
I 101% respect that. It will be changed first thing. As for the performant part: We are talking data sets of what? 3-10 nics ;) I don't think it really matter here. I prefer ToList as I think it is more flexible to work with. Anyway, I was intrigued by your argument, so I googled [C# tolist vs toarray](https://www.google.dk/search?q=c%23+tolist+vs+toarray&oq=C%23+ToList+vs+T&aqs=chrome.0.0j69i58j69i57.5955j0j7&sourceid=chrome&ie=UTF-8) and this was my top hit:
https://stackoverflow.com/questions/1105990/is-it-better-to-call-tolist-or-toarray-in-linq-queries

But that is only trivia. The important thing here is that you prefer ToArray, so ToArray it is ;)